sessionctx/variable/session.go

```diff
@@ -1305,6 +1305,11 @@ func (s *SessionVars) InTxn() bool {
 	return s.GetStatusFlag(mysql.ServerStatusInTrans)
 }
 
+// LocalTemporaryTableExists returns if there are local temporary tables in the session.
+func (s *SessionVars) LocalTemporaryTableExists() bool {
+	return s.LocalTemporaryTables != nil
```

Comment:
Good!
Expose those internal field of a struct is not under control, user can do anything with it.
When we refactor, there is no single places for all, and missing some result in bugs.